TransIT.API/Controllers/MalfunctionGroupController.cs

```diff
@@ -1,47 +1,82 @@
 namespace TransIT.API.Controllers
 {
+    [ApiController]
+    [EnableCors("CorsPolicy")]
+    [Produces("application/json")]
+    [Route("api/v1/[controller]")]
     [Authorize(Roles = "ADMIN,ENGINEER,ANALYST")]
-    public class MalfunctionGroupController : DataController<MalfunctionGroup, MalfunctionGroupDTO>
+    public class MalfunctionGroupController : FilterController<MalfunctionGroupDTO>
     {
-        private readonly IMalfunctionGroupService _malfunctionGroupService;
+        private readonly IServiceFactory _serviceFactory;
 
-        public MalfunctionGroupController(
-            IMapper mapper, 
-            IMalfunctionGroupService malfunctionGroupService,
-            IFilterService<MalfunctionGroup> odService
-            ) : base(mapper, malfunctionGroupService, odService)
+        public MalfunctionGroupController(IServiceFactory serviceFactory, IFilterServiceFactory filterServiceFactory)
+            : base(filterServiceFactory)
         {
-            _malfunctionGroupService = malfunctionGroupService;
+            _serviceFactory = serviceFactory;
+        }
+
+        [HttpGet]
+        public virtual async Task<IActionResult> Get([FromQuery] uint offset = 0, uint amount = 1000)
+        {
+            var result = await _serviceFactory.MalfunctionGroupService.GetRangeAsync(offset, amount);
+            return result != null
+                ? Json(result)
+                : (IActionResult)BadRequest();
+        }
+
+        [HttpGet("{id}")]
+        public virtual async Task<IActionResult> Get(int id)
+        {
+            var result = await _serviceFactory.MalfunctionGroupService.GetAsync(id);
+            return result != null
+                ? Json(result)
+                : (IActionResult)BadRequest();
+        }
+
+        [HttpGet("/search")]
+        public virtual async Task<IActionResult> Get([FromQuery] string search)
+        {
+            var result = await _serviceFactory.MalfunctionGroupService.SearchAsync(search);
+            return result != null
+                ? Json(result)
+                : (IActionResult)BadRequest();
         }
 
         [HttpPost]
         [Authorize(Roles = "ADMIN")]
-        public override Task<IActionResult> Create([FromBody] MalfunctionGroupDTO obj)
+        public async Task<IActionResult> Create([FromBody] MalfunctionGroupDTO obj)
         {
-            return base.Create(obj);
+            var createdEntity = await _serviceFactory.MalfunctionGroupService.CreateAsync(obj);
+            return createdEntity != null
+                ? CreatedAtAction(nameof(Create), createdEntity)
+                : (IActionResult)BadRequest();
         }
 
         [HttpPut("{id}")]
         [Authorize(Roles = "ADMIN")]
-        public override Task<IActionResult> Update(int id, [FromBody] MalfunctionGroupDTO obj)
+        public async Task<IActionResult> Update(int id, [FromBody] MalfunctionGroupDTO obj)
         {
-            return base.Update(id, obj);
+            obj.Id = id;
```

Comment:
should not be like that, pass obj.id from front-end, can be done in a separate task, not in this pull request